File: lib/ts/src/MockMsgPactSchema.ts
import { VFieldDeclaration } from './internal/types/VFieldDeclaration';
import { VType } from './internal/types/VType';
import { createMockMsgPactSchemaFromFileJsonMap } from './internal/schema/CreateMockMsgPactSchemaFromFileJsonMap';
import { getSchemaFileMap } from './internal/schema/GetSchemaFileMap';
import { FsModule, PathModule } from './fileSystem';

export class MockMsgPactSchema {
    /**
     * A parsed msgPact schema.
     */

    original: any[];
    full: any[];
    parsed: Record<string, VType>;
    parsedRequestHeaders: Record<string, VFieldDeclaration>;
    parsedResponseHeaders: Record<string, VFieldDeclaration>;

    constructor(
        original: any[],
        full: any[],
        parsed: Record<string, VType>,
        parsedRequestHeaders: Record<string, VFieldDeclaration>,
        parsedResponseHeaders: Record<string, VFieldDeclaration>,
    ) {
        this.original = original;
        this.full = full;
        this.parsed = parsed;
        this.parsedRequestHeaders = parsedRequestHeaders;
        this.parsedResponseHeaders = parsedResponseHeaders;
    }

    static fromJson(json: string): MockMsgPactSchema {
        return createMockMsgPactSchemaFromFileJsonMap({ auto_: json });
    }

    static fromFileJsonMap(jsonDocuments: Record<string, string>): MockMsgPactSchema {
        return createMockMsgPactSchemaFromFileJsonMap(jsonDocuments);
    }

    static fromDirectory(directory: string, fs: FsModule, path: PathModule): MockMsgPactSchema {
        const m = getSchemaFileMap(directory, fs, path);
        return createMockMsgPactSchemaFromFileJsonMap(m);
    }
}

File: lib/java/src/main/java/io/github/msgpact/MockMsgPactSchema.java
package io.github.msgpact;

import static io.github.msgpact.internal.schema.CreateMockMsgPactSchemaFromJsonDocuments.createMockMsgPactSchemaFromFileJsonMap;
import static io.github.msgpact.internal.schema.GetSchemaFileMap.getSchemaFileMap;

import java.util.List;
import java.util.Map;

import io.github.msgpact.internal.types.VFieldDeclaration;
import io.github.msgpact.internal.types.VType;

/**
 * A parsed msgPact schema.
 */
public class MockMsgPactSchema {

    public final List<Object> original;
    public final Map<String, VType> parsed;
    public final Map<String, VFieldDeclaration> parsedRequestHeaders;
    public final Map<String, VFieldDeclaration> parsedResponseHeaders;

    public MockMsgPactSchema(List<Object> original,
            Map<String, VType> parsed,
            Map<String, VFieldDeclaration> parsedRequestHeaders,
            Map<String, VFieldDeclaration> parsedResponseHeaders) {
        this.original = original;
        this.parsed = parsed;
        this.parsedRequestHeaders = parsedRequestHeaders;
        this.parsedResponseHeaders = parsedResponseHeaders;
    }

    public static MockMsgPactSchema fromJson(String json) {
        return createMockMsgPactSchemaFromFileJsonMap(Map.of("auto_", json));
    }

    public static MockMsgPactSchema fromFileJsonMap(Map<String, String> fileJsonMap) {
        return createMockMsgPactSchemaFromFileJsonMap(fileJsonMap);
    }

    public static MockMsgPactSchema fromDirectory(String directory) {
        final var map = getSchemaFileMap(directory);
        return createMockMsgPactSchemaFromFileJsonMap(map);
    }
}
